src/NLog/Targets/DatabaseParameterInfo.cs

```diff
@@ -98,6 +98,20 @@ public DatabaseParameterInfo(string parameterName, Layout parameterLayout)
         /// <docgen category='Parameter Options' order='10' />
         [DefaultValue(0)]
         public byte Scale { get; set; }
+
+        /// <summary>
+        /// Gets or sets the column type of the database.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the database column type property.
+        /// </summary>
+        /// <docgen category='Parameter Options' order='10' />
+        [DefaultValue(null)]
+        public string DbTypePropertyName { get; set; }
```

Comment:
True it is a bit magic and reflection does impact performance. I can change it up to just hard code the needed value for just Oracle for now like you did in your pseudo code, Then in the future a code change will be required to support any other special cases like this. I'll update my code in a day or two when I next get the chance.